test/integration/metrics_service_integration_test.cc

```diff
@@ -59,6 +58,34 @@ class MetricsServiceIntegrationTest : public HttpIntegrationTest,
     EXPECT_STREQ("application/grpc",
                  metrics_service_request_->headers().ContentType()->value().c_str());
     EXPECT_TRUE(request_msg.envoy_metrics_size() > 0);
+    const Protobuf::RepeatedPtrField<::io::prometheus::client::MetricFamily>& envoy_metrics =
+        request_msg.envoy_metrics();
+    bool known_counter_exists = false;
+    bool known_gauge_exists = false;
+    int known_counter_value = -1;
+    int known_gauge_value = -1;
+    std::string known_counter("cluster.metrics_service.membership_change");
+    std::string known_gauge("cluster.metrics_service.membership_total");
+    int metrics_size = envoy_metrics.size();
+    for (int i = 0; i < metrics_size; i++) {
```

Comment:
nit: You should be able to use a c++11 for loop here and just iterate through the `envoy_metrics()` metrics.